src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1647,16 +1998,18 @@ public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, T
                 String workload = k8sObject.getName();
                 String workload_type = k8sObject.getType();
                 HashMap<String, ContainerData> containerDataMap = k8sObject.getContainerDataMap();
-                // Iterate over containers
-                for (Map.Entry<String, ContainerData> entry : containerDataMap.entrySet()) {
-                    ContainerData containerData = entry.getValue();
-                    String containerName = containerData.getContainer_name();
-                    if (null == interval_end_time) {
-                        LOGGER.info(KruizeConstants.APIMessages.CONTAINER_USAGE_INFO);
-                        String queryToEncode = null;
-                        if (null == maxDateQuery || maxDateQuery.isEmpty()) {
-                            throw new NullPointerException("maxDate query cannot be empty or null");
-                        }
+                // check if containerDataMap is not empty
+                if (!containerDataMap.isEmpty()) {
+                    // Iterate over containers
+                    for (Map.Entry<String, ContainerData> entry : containerDataMap.entrySet()) {
+                        ContainerData containerData = entry.getValue();
+                        String containerName = containerData.getContainer_name();
+                        if (null == interval_end_time) {
+                            LOGGER.info(KruizeConstants.APIMessages.CONTAINER_USAGE_INFO);
+                            String queryToEncode = null;
+                            if (null == maxDateQuery || maxDateQuery.isEmpty()) {
+                                throw new NullPointerException("maxDate query cannot be empty or null");
+                            }
 
 
                             LOGGER.info("maxDateQuery: {}", maxDateQuery);
```

Comment:
Change this to `debug` or remove it since we're already logging `dateMetricsUrl` below

Reply:
Updated to use debug.